toolchain/lower/file_context.h

```diff
@@ -91,16 +91,19 @@ class FileContext {
 
   // Maps callables to lowered functions. SemIR treats callables as the
   // canonical form of a function, so lowering needs to do the same.
-  llvm::SmallVector<llvm::Function*> functions_;
+  // We resize this directly to the correct size rather than using a small size.
```

Comment:
Isn't this a little misleading about the behavior? If a file has something like 64 functions or types, wouldn't it avoid a heap allocation?

FWIW, maybe:

```suggestion
  // We resize this directly to the correct size, and don't try to fit into the same allocation.
```

or just:

```suggestion
  // We resize this directly to the (often large) correct size.
```

(and below)

Reply:
I meant "small size" as in the "small size optimization" -- I don't think that's misleading, but I can see how you might instead read it as "a size that's too small for the content" or something like that. I'll use the latter suggestion.